syntax = "proto3";

package message;

enum VisitorType {
	unknown = 0;
	publisher = 1;
	subscriber = 2;
}

message VisitorProperty {
	VisitorType type = 1;
	string category = 2;
	string topic = 3;
	string extension = 4;
}

message PublisherMessage {
	VisitorProperty properties = 1;
	string payload = 2;
}

message SubscriberMessage {
	VisitorProperty properties = 1;
}